interface/src/app/gene/network/network.js

```diff
@@ -157,9 +157,9 @@ angular.module('adage.gene.network', [
       var refreshSlider = function() {
         $timeout(function() {
           $scope.$broadcast('rzSliderForceRender');
-        }, 250);
-        // Note by dhu: 250ms is an arbitrary timeout value. The minimum timeout
-        // value that worked on my desktop is 150ms.
+        }, 1000);
+        // Note by dhu: 1000ms is an arbitrary timeout value. The minimum
+        // timeout value that worked on my desktop is 150ms.
```

Comment:
This fixes the slider glitch for me, too. Nice!

The fact that it works for a longer timeout but not for a shorter one is an indication that there is a race condition here. Some event must happen before issuing the `rzSliderForceRender` and it happens sometimes within 250ms and sometimes not. If we knew what that event was we would be able to make a 100% reliable fix. If we don't know what the event is, then this solution is a good enough workaround for me.

Reply:
:+1:

Reply:
@mhuyck What you said about race condition is a good point. I took a look at the source code of `rzslider`. Unsurprisingly the event is related to a lot of DOM manipulations. I tried a few quick fixes but they didn't work. The fact that its README.md suggests using `$timeout()` probably indicates that that is the best solution.

I will add a new issue to keep track of it though.